### PR TITLE
feat: rust binary releases

### PR DESCRIPTION
**Pull request by jcgruenhage**
Based on sammys previous work, mostly adjusted to work for crates where there's multiple binaries, or where the binary target has a different name from the package.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    name: "Build"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Prepare dependencies
+        run: |
+          rustup update stable
```
Comment (jdreichmann) [inline .github/workflows/rust.yml:17]: Where does it make sure `rustup` is available? `runs-on: ubuntu-latest` does not imply rustup is present (to me)

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+      - name: Build
+        run: cargo auditable build --release
+
+      - name: Create release
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          gh release create ${{ github.ref_name }} ${{ steps.crate_metadata.outputs.binary_targets }}
+        if: github.ref_type == 'tag'
```
Comment (jdreichmann) [inline .github/workflows/rust.yml:40]: That means every tag, is restricting to `vX.Y.z` an easy thing to do?

File: .github/workflows/rust.yml (hunk 1 above)
Comment (jcgruenhage) [reply .github/workflows/rust.yml:17]: https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md

They ship everything under the sun in terms of development tooling

File: .github/workflows/rust.yml (hunk 2 above)
Comment (jcgruenhage) [reply .github/workflows/rust.yml:40]: not really, unless we also want the rest to not run on other tags

**Review by jdreichmann — APPROVED** [review]

**Pull request merged**